.github/copilot-instructions.md

```diff
@@ -0,0 +1,173 @@
+- **`registry_tools/global-validator.py`** - Cross-file validation
+- **`registry_tools/GEDCOM.io/yaml-schema.yaml`** - Main validation schema (submodule)
+
+## Common Tasks
+
+### Adding or Modifying YAML Files
+
+1. Create or edit YAML file in appropriate directory following naming convention
+2. Run schema validation: `cd registry_tools && python3 validator.py ../path/to/file.yaml`
+3. Run complete validation workflow (see above)
+4. Run generation scripts to update summary files
+5. Always commit both the YAML changes and regenerated files
+
+### YAML File Structure
+
+All YAML files must follow the basic structure defined in the official documentation:
+
+- **YAML File Format**: https://github.com/FamilySearch/GEDCOM.io/blob/main/_pages/yaml-file-format.md
+- **Schema for v7 files**: https://github.com/FamilySearch/GEDCOM.io/blob/main/yaml-schema.yaml  
+- **Schema for v5.5.1 files**: https://github.com/FamilySearch/GEDCOM.io/blob/main/yaml-schema-v5.5.1.yaml
+
+Key required fields for all YAML files:
+- `lang`: Language tag (e.g., "en-US")
+- `type`: One of "structure", "enumeration", "enumeration set", "calendar", "month", "data type", "uri"
+- `uri`: The URI that identifies the concept
+- `specification`: List of descriptions (required for most types except "enumeration set")
+
+### File Naming Convention
+
+Files are organized as: `type/subtype/name.yaml` where:
+- **type**: Value of YAML file's `type:` key (spaces → hyphens)
+- **subtype**: `standard` (official GEDCOM 7 spec) or `extension` (community extensions)
+- **name**: Identifier selected at registration time
+
+### Git Submodules
+
+The repository uses two git submodules:
+- **`registry_tools/GEDCOM`** - FamilySearch GEDCOM specification
+- **`registry_tools/GEDCOM.io`** - GEDCOM.io schemas and documentation
+
+ALWAYS run `git submodule update --init --recursive` after cloning.
+
+## Time Expectations
+
+- **Submodule initialization**: 5 seconds - NEVER CANCEL
+- **YAML syntax validation**: 2 seconds - NEVER CANCEL
+- **Schema validation**: 9 seconds - NEVER CANCEL, set timeout 60+ seconds
+- **Global validation**: 1 second - NEVER CANCEL
+- **File generation**: 5 seconds total - NEVER CANCEL, set timeout 30+ seconds
+- **Complete validation workflow**: 18 seconds - NEVER CANCEL, set timeout 60+ seconds
+
+## Common Command Reference
+
+```bash
+# Complete fresh setup
+git submodule update --init --recursive
+
+# Validate single file
+cd registry_tools && python3 validator.py ../structure/standard/record-INDI.yaml
+
+# Run complete validation (like CI)
+yamllint . && cd registry_tools && find .. -type f -name "*.yaml" -not -path "../registry_tools/*" -exec dirname {} \; | sort -u | while read -r dir; do yaml_files=("$dir"/*.yaml); if [ "${#yaml_files[@]}" -gt 0 ]; then echo "Checking directory $dir"; python3 validator.py "${yaml_files[@]}"; if [ $? -ne 0 ]; then exit 1; fi; fi; done && cd .. && python3 ./registry_tools/global-validator.py
```

Comment:
```suggestion
yamllint . && python3 ./registry_tools/global-validator.py
```

As noted above, the loop over individual files is not needed because global_valdiator does this more efficiently already.